Systems/Barrel/BlockBarrel.cs

```diff
@@ -345,7 +350,31 @@ public override void OnLoaded(ICoreAPI api)
                 capacityLitresFromAttributes = Attributes["capacityLitres"].AsInt(50);
             }
 
+            if (Attributes?["emptyShape"].Exists == true)
+            {
+                emptyShape = AssetLocation.Create(Attributes["emptyShape"].AsString("shapes/block/wood/barrel/empty.json")).WithPathPrefixOnce("shapes").WithPathAppendixOnce(".json");
+            }
 
+            if (Attributes?["sealedShape"].Exists == true)
+            {
+                sealedShape = AssetLocation.Create(Attributes["sealedShape"].AsString("shapes/block/wood/barrel/closed.json")).WithPathPrefixOnce("shapes").WithPathAppendixOnce(".json");
+            }
+            
+            if (Attributes?["contentsShape"].Exists == true)
+            {
+                contentsShape = AssetLocation.Create(Attributes["contentsShape"].AsString("shapes/block/wood/barrel/contents.json")).WithPathPrefixOnce("shapes").WithPathAppendixOnce(".json");
+            }
+                        
+            if (Attributes?["opaqueLiquidContentsShape"].Exists == true)
+            {
+                contentsShape = AssetLocation.Create(Attributes["opaqueLiquidContentsShape"].AsString("shapes/block/wood/barrel/opaqueliquidcontents.json")).WithPathPrefixOnce("shapes").WithPathAppendixOnce(".json");

```

Comment:
wrong field name here, and this code is indeed a bit suboptimal. will fix.

Reply:
domain field also not correctly set

Reply:
Correct solution
![image](https://github.com/user-attachments/assets/38aa31e5-9625-430a-8271-347274099328)